--- /etc/nginx/sites-available/elasticbeanstalk-nginx-docker-proxy.conf.org
+++ /etc/nginx/sites-available/elasticbeanstalk-nginx-docker-proxy.conf
@@ -22,9 +22,9 @@
             set $hour $4;
         }
         access_log /var/log/nginx/healthd/application.log.$year-$month-$day-$hour healthd;
 
-        access_log    /var/log/nginx/access.log;
+        access_log /var/log/nginx/access.log combined_extra;
     
         location / {
             proxy_pass            http://docker;
             proxy_http_version    1.1;
