--- /etc/nginx/sites-available/elasticbeanstalk-nginx-docker-proxy.conf.org
+++ /etc/nginx/sites-available/elasticbeanstalk-nginx-docker-proxy.conf
@@ -6,8 +6,25 @@
     map $http_X_Request_Start $request_start {
         default "$msec";
         ~. "$http_X_Request_Start";
     }
+
+    map $http_x_forwarded_for $is_forwarded {
+        default "1";
+        "" "0";
+    }
+
+    map $http_host $is_bad_host {
+        default "0";
+        "" "1";
+        ~^[0-9.]+$ "1";
+        ~\.amazonaws\.com$ "1";
+    }
+
+    map "$is_forwarded:$is_bad_host" $reject_bad_host {
+      default "0";
+      "1:1" "1";
+    }
     
     server {
         listen 80;
 
@@ -23,8 +40,12 @@
         }
         access_log /var/log/nginx/healthd/application.log.$year-$month-$day-$hour healthd;
 
         access_log /var/log/nginx/access.log combined_extra;
+
+        if ($reject_bad_host) {
+          return 403;
+        }
     
         location / {
             proxy_pass            http://docker;
             proxy_http_version    1.1;
